src/mobile/src/main/java/tupperdate/android/onboarding/Onboarding.kt

```diff
@@ -28,39 +28,11 @@ import tupperdate.android.ui.material.BrandedButton
 import tupperdate.android.ui.material.BrandedTitleText
 import tupperdate.api.AuthenticationApi
 import tupperdate.api.api
-import kotlin.String
-import kotlin.Unit
-import kotlin.let
 
-private enum class Error {
-    Internal,
-    InvalidNumber,
-}
-
-/* We could use a sealed class and store the error inside
- * a special type, but Kotlin won't let us remember the State,
- * only the implemented subclass... Something like this:
- *
- * private sealed class State()
- * private data class Error(val error: String) : State()
- * private object Pending : State()
- * private object WaitingForInput : State()
- *
- * When this can be achieved, we can remove the redundant error variable.
- */
-private enum class State {
-    Error,
-    WaitingForInput,
-    Pending,
-}
-
-@Composable
-private fun getErrorString(error: Error): String {
-    return when (error) {
-        Error.Internal -> stringResource(R.string.onboarding_requestCode_error_internal)
-        Error.InvalidNumber -> stringResource(R.string.onboarding_requestCode_error_invalid_number)
-    }
-}
+private sealed class State()
+private data class Error(val error: Int) : State()
```

Comment:
Thanks 😄 This might already be a bit better :

```suggestion
private data class Error(@StringRes val error: Int) : State()
```

While it's definitely open to discussion, I wonder if we'd not be better of with an `enum class State` with 5 cases, including `InternalError` and `InvalidNumberError`, or, at least, have an `ErrorMessage` enum with the different error cases.

Storing the string (or string resource `Int`) representation of the error rather than having a proper type representing it seems a bit off to me.

Reply:
I do not think that having a State with 5 cases is a good idea. I can revert the `error: Int` to have it's own type, like it was before. (I don't remember why I removed it...)